sdks/js/src/types.ts

```diff
@@ -1844,6 +1872,43 @@ export type GetWorkspaceUsageRequestType = z.infer<
   typeof GetWorkspaceUsageRequestSchema
 >;
 
+export const FileUploadUrlRequestSchema = z.object({
+  contentType: z.string(),
+  fileName: z.string(),
+  fileSize: z.number(),
+  useCase: z.union([z.literal("conversation"), z.literal("avatar")]),
+});
+
+export const FileTypeSchema = z.object({
+  contentType: z.string(),
+  downloadUrl: z.string().optional(),
+  fileName: z.string(),
+  fileSize: z.number(),
+  id: z.string(),
+  status: z.enum(["created", "failed", "ready"]),
+  uploadUrl: z.string().optional(),
+  publicUrl: z.string().optional(),
+  useCase: z.enum(["conversation", "avatar", "tool_output"]),
```

Comment:
In schema defining responses, (such as the `status` and `useCase` enums here), use `FlexibleEnumSchema`, that way, it won't break if a "new" value is added by us.

(note: You'll have to declare them separately for some typechecking reason that I didn't understand)

Reply:
I don't have any typing error when declaring it inline, I will try like that ?

Reply:
hum, is the infered type "string" or an enum ?

Reply:
you're right, very weird